File: Data Structures & Algorithms/binary-tree-right-side-view/submission-0.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right

class Solution:
    def rightSideView(self, root: Optional[TreeNode]) -> List[int]:
        queue = deque()
        rightList = []

        if root:
            queue.append(root)

        while len(queue) > 0:
            for i in range(len(queue)):
                curr = queue.popleft()
                if i == 0:
                    rightList.append(curr.val)
                if curr.right:
                    print("go right")
                    queue.append(curr.right)
                if curr.left:
                    print("right was empty, go left")
                    queue.append(curr.left)
        
        return rightList

File: Data Structures & Algorithms/kth-smallest-integer-in-bst/submission-0.go
/**
 * Definition for a binary tree node.
 * type TreeNode struct {
 *     Val int
 *     Left *TreeNode
 *     Right *TreeNode
 * }
 */

func kthSmallest(root *TreeNode, k int) int {
    var outVal int
	var dfs func(*TreeNode)

	dfs = func(root *TreeNode) {
		if root == nil || k == 0 {
			return
		}

		dfs(root.Left)
		k--
		if k == 0 {
			outVal = root.Val
			return
		}

		dfs(root.Right)
	}

	dfs(root)
	return outVal
}
